subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskOptionsSupplier.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.gradle.api.internal.tasks.options.BooleanOptionElement;
+import org.gradle.api.internal.tasks.options.BuiltInOptionElement;
+import org.gradle.api.internal.tasks.options.InstanceOptionDescriptor;
+import org.gradle.api.internal.tasks.options.OptionDescriptor;
+import org.gradle.api.internal.tasks.options.OptionElement;
+import org.gradle.api.internal.tasks.options.OptionReader;
+import org.gradle.api.specs.Specs;
+import org.gradle.internal.Pair;
+import org.gradle.util.internal.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class is responsible for supplying the built-in options and
+ * the {@link BooleanOptionElement mutually exclusive options} of a task,
+ * based on the task options provided by the {@link OptionReader}.
+ */
+public class TaskOptionsSupplier {
```

Comment:
💭 I'd consider renaming this type, as it has a `get` method and the name has a `Supplier` suffix, but it is completely unrelated to `java.util.function.Supplier`.

Maybe `TaskOptionsCreator`?  `TaskOptionsGenerator`?  With the `get` method also renamed to `create`/`generate`/`for`?